### PR TITLE
Invalid date checks

### PR DESCRIPTION
**Pull request by aramk**
Firstly, this is a very useful library! I ran into an issue where I incorrectly parsed a date and passed it into `between()`, which caused an infinite loop since it treated the date similar to `undefined`. While we may expect the iterator to be called indefinitely (until we return a falsy result) when we provide a falsy before date, it's likely an error on our part if we pass in an `Invalid Date`. I've added a check in `init()` to fail on that along with some tests. I also added two small methods for checking if an object is a date and whether it's valid.

### CODE REVIEW

File: lib/rrule.js [hunk 1]
```diff
@@ -217,6 +217,14 @@ var dateutil = {
             bits[6] || 0,
             bits[7] || 0
         ));
+    },
+
+    isDate: function (date) {
+        return date instanceof Date;
+    },
+
+    isValid: function (date) {
```
Comment (jakubroztocil) [inline lib/rrule.js]: I think a better name would be `isValidDate()`

Comment (aramk) [inline lib/rrule.js]: Done

**Comment by arolson101** [comment]
Hi!  I'm trying to clear out bugs and PRs.  If this PR is still wanted/relevant, please comment and/or fix the conflicts!

**Comment by davidgoli** [comment]
This PR is too out of date to merge at this point, however, the functionality is still welcome in a new PR!

**Pull request closed without merging**